### ISSUE TITLE
Failure during CosmosDB creation

### ISSUE THREAD

**Issue by danielrbradley** [pulumi/pulumi-azure#2434]
The `Test_cosmosdb` upgrade test has been disabled due to a persistent failure during the initial deployment.

https://github.com/pulumi/pulumi-azure/blob/c9be11ddb3490dd973bca57ed5aa25cf4db04511/provider/provider_yaml_test.go#L88-L91

### Pulumi YAML Program

```yaml
resources:
  rg:
    type: azure:core:ResourceGroup
    properties:
      location: westus

  db:
    type: azure:cosmosdb:Account
    properties:
      location: ${rg.location}
      resourceGroupName: ${rg.name}
      offerType: Standard
      kind: MongoDB
      automaticFailoverEnabled: true
      capabilities:
        - name: EnableAggregationPipeline
        - name: mongoEnableDocLevelTTL
        - name: MongoDBv3.4
        - name: EnableMongo
      consistencyPolicy:
        consistencyLevel: BoundedStaleness
        maxIntervalInSeconds: 300
        maxStalenessPrefix: 100000
      geoLocations:
        - location: westus
          failoverPriority: 0
```

The error message reports:
```
            Status: "ServiceUnavailable"
            Code: ""
            Message: "Database account was created successfully, but the following regions failed to be added to account: < br/ >East US : Operation Id : xxx, Error : Message: 
                        { "code":"ServiceUnavailable",
                          "message":"Sorry, we are currently experiencing high demand in East US region, and cannot fulfill your request at this time Tue, 24 Sep 2024 09:51:02 GMT. To request region access for your subscription, please follow this link https://aka.ms/cosmosdbquota for more details on how to create a region access request.\r\nActivityId: xxx, Microsoft.Azure.Documents.Common/2.14.0\"
                        },
                        Request URI: /serviceReservation,
                        RequestStats: , SDK: Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documen
ts.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0 < br/ >, Microsoft.Azure.Documents.Common/2.14.0"
```

However, changing the region does not seem to fix the issue as it occurs everywhere. There is also no service issues reported at the time of recording.
